src/components/Vote.jsx

```diff
@@ -0,0 +1,103 @@
+import { useEffect, useState } from "react";
+import axios from "axios";
+import '../style-sheets/Vote.css'
+import { useContext } from 'react';
+import { UserContext } from '../contexts/User';
+
+function Vote({vote, article_id}){
+    const [voteState, setVoteState] = useState();
+    const [err, setErr] = useState(null);
+    const { user, setUser } = useContext(UserContext);
+
+    useEffect(() => {
+        setVoteState(vote)
+    }, [vote])
+
+    function checkIfVoted(voteType){
+        if(user.votes.hasOwnProperty(article_id)){
+            if(user.votes[article_id] === voteType){
+                return false;
+            }
+            return true;
+        }
+        return true;
+    }
+
+    function undoVote(voteType){
+        if(user.votes.hasOwnProperty(article_id)){
+            if(user.votes[article_id] === voteType){
+                return true;
+            }
+            return false;
+        }
+        return false;
+    }
+
+    //optimistic rendering of vote incremement
+    function handleVote(event) {
+        let votes = 0
+        event.preventDefault();
+        if(event.target.value === "upvote" && checkIfVoted("upvote")){
+            if(undoVote("downvote")){
+                votes = 2
+            }
```

Comment:
This functionality is a little strange. Assuming we start at 0 votes it is possible to vote up to 1 or downvote to -1, but i think we should also be able to remove our vote (ie vote up to 1 and then change our mind and go back to 0), but this isn't possible